### PR TITLE
ARP: implement decoder and logger v6

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6827

Describe changes:
- Rebase
- Check if first byte is != '\0' instead of using strlen

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1807

### CODE REVIEW

**Review by catenacyber — COMMENTED** [review]
Thanks for the work Giuseppe

CI : 🟢 
Code : Checking now
Commits segmentation : good
Commit messages : good, I wonder if `output/json: check 5-tuple values prior to logging` deserves its own ticket as a breaking change in suricata output that some tools may rely on...
Git ID set : looks fine for me
CLA : you already contributed
Doc update : looks good
Redmine ticket : ok, updated to reference this PR
Rustfmt : no rust
Tests : SV looks good
Dependencies added: none added

File: doc/userguide/output/eve/eve-json-format.rst [hunk 1]
```diff
@@ -3011,3 +3011,47 @@ Example of DHCP log entry (extended logging enabled):
     "client_id":"54:ee:75:51:e0:66",
     "dns_servers":["192.168.1.50","192.168.1.49"]
   }
+
+Event type: ARP
+-----------------
```
Comment (catenacyber) [inline doc/userguide/output/eve/eve-json-format.rst:3016]: @jufajardini should there be some less dashes ? To match the length of `Event type: ARP` ?

File: doc/userguide/output/eve/eve-json-output.rst [hunk 2]
```diff
@@ -266,6 +266,20 @@ enabled, then the log gets more verbose.
...
+
+ARP records are logged as one entry for the request, and one entry for
+the response.
+
+YAML::
+
+        - arp:
+            enabled: no
+
+The logger is disabled by default since ARP can generate a large
```
Comment (catenacyber) [inline doc/userguide/output/eve/eve-json-output.rst:280]: Can thresholding help there ?

File: src/output-json-arp.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Giuseppe Longo <giuseppe@glongo.it>
+ */
+#ifndef __OUTPUT_JSON_ARP_H__
```
Comment (catenacyber) [inline src/output-json-arp.h:23]: We have newer guard style

File: src/output-json-arp.c [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    }
+}
+
+static const char *PrototypeToString(uint16_t proto)
+{
+    switch (proto) {
+        case 0x0800:
+            return "ipv4";
+        default:
+            return "unknown";
```
Comment (catenacyber) [inline src/output-json-arp.c:148]: Could we have the value like `unknown-0x08001` ?

File: src/output-json-arp.c [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+#include "threads.h"
+#include "tm-threads.h"
+#include "threadvars.h"
+#include "util-debug.h"
+
+#include "decode-ipv4.h"
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "detect-engine-mpm.h"
```
Comment (catenacyber) [inline src/output-json-arp.c:39]: Do we need mpm here ?

File: src/output-json.h [hunk 6]
```diff
@@ -50,6 +50,7 @@ typedef struct JsonAddrInfo_ {
     Port sp;
     Port dp;
     char proto[JSON_PROTO_LEN];
+    bool log_port;
```
Comment (catenacyber) [inline src/output-json.h:53]: Cannot it be within a reduced scope, not inside a struct, but just as a variable in `CreateEveHeader` where we have the Packet as argument ?

File: src/output-json-arp.c [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+            return "op_exp2";
+        default:
+            return "unknown";
+    }
+}
+
+static int JsonArpLogger(ThreadVars *tv, void *thread_data, const Packet *p)
+{
+    OutputJsonThreadCtx *thread = thread_data;
+    char srcip[46] = "";
```
[the remark this answers is not in the document]
Comment (glongo) [reply src/output-json-arp.c:216]: It comes from here:

```
void JsonAddrInfoInit(const Packet *p, enum OutputJsonLogDirection dir, JsonAddrInfo *addr)
{
    char srcip[46] = {0}, dstip[46] = {0};
    ...
}
```

File: src/output-json-arp.c (hunk 4 above)
Comment (glongo) [reply src/output-json-arp.c:148]: We could, if we wanted to

File: src/output-json.h (hunk 6 above)
Comment (glongo) [reply src/output-json.h:53]: Yes, it could be an alternative.

File: src/output-json-arp.c (hunk 7 above)
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/output-json-arp.c:216]: Could reuse `JSON_ADDR_LEN` ;-)

File: doc/userguide/output/eve/eve-json-format.rst (hunk 1 above)
Comment (jufajardini) [reply doc/userguide/output/eve/eve-json-format.rst:3016]: Indeed!

**Comment by glongo** [comment]
Replaced with https://github.com/OISF/suricata/pull/11005

**Pull request closed without merging**